### DISCUSSION TITLE
SD3 cannot finetunes a better model (hand and face deformation)?

### DISCUSSION THREAD

**Discussion by KaiWU5** [huggingface/diffusers#8878] [Q&A]
### Describe the bug

I want to finetune sd3 to improve its human generation quality with 3million high-quality human datasets (which has been proven useful on sdxl and other models).  But hand and face deformation doesn't improve much after two days of training. 

I am using [train](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_sd3.py) script

What I have been done so far:
1. regular training with 3 million data with batch size 2x24(V100) for 2 epochs with lr 5e-6 and adamw optimizer
2. prodigy optimizer training  with same setting
3. Add q,k RMS norm to each attention layer
4. only train several blocks

All of my training gives me nearly the same deformation results, where the hands are never normal like human. 

Could you some provide more experiments about sd3 training? There seems no easy way to adapt sd3 for human generation

### Reproduction

Has described in bug part

### System Info

V100 24GPU, batchsize 2 for each card, 3 million human data with aesthetic score > 4.5

**Comment by DN6** [comment]
Hi @KaiWU5 I think this question would be better to ask in the Discussions section.

**Comment by mliand** [comment]
You can show me your loss training

**Comment by heart-du** [comment]
I have the same question.

**Comment by KaiWU5** [comment]
<img width="364" alt="image" src="https://github.com/huggingface/diffusers/assets/38367713/7ece99a1-b9d4-4836-b4e9-236be605be48">
Here's my loss curve. The loss after adding qknorm is similar. Thanks for noticing and I will continue discussing the results in the discussion section.

**Reply by xduzhangjiayu** [reply]
> <img alt="image" width="364" src="https://private-user-images.githubusercontent.com/38367713/347733695-7ece99a1-b9d4-4836-b4e9-236be605be48.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NTM5OTgsIm5iZiI6MTcyODY1MzY5OCwicGF0aCI6Ii8zODM2NzcxMy8zNDc3MzM2OTUtN2VjZTk5YTEtYjlkNC00ODM2LWI0ZTktMjM2YmU2MDViZTQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDEzMzQ1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjYmZjNGMzZDc2MDdjYjUyZTNjZGEyZDY2NjU5NzUyM2ViMzMwZGNjNWRiZTg2OTBhNzRlNjE0NTBkMzc0N2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.01SKCVR8csIOaXj0Ccq0YBMUBfYusKf9hvg7gkew3rg"> Here's my loss curve. The loss after adding qknorm is similar. Thanks for noticing and I will continue discussing the results in the discussion section.

Your loss curve is same as I trained sd3 controlnet, have you solve this issue or do you have any suggestions? Thanks!

**Comment by appleyang123** [comment]
+1, I have the same issue.

**Reply by CodeExplode** [reply]
Have you tried training SD3.5 rather than SD3? The original model didn't ship with some RMS Norm layers which may have been important for stabilizing training.

I found that when training SD3 (the original), I got much better results just training the context path, specifically the post-attention parameters of the context path. It seems these models can 'draw' almost anything, it's just a matter of finding the correct way to speak to them to ask, and training the context path changes how it processes the prompt into a format which the rest of the model can understand and respond to.

**Reply by xduzhangjiayu** [reply]
> Have you tried training SD3.5 rather than SD3? The original model didn't ship with some RMS Norm layers which may have been important for stabilizing training.
> 
> I found that when training SD3 (the original), I got much better results just training the context path, specifically the post-attention parameters of the context path. It seems these models can 'draw' almost anything, it's just a matter of finding the correct way to speak to them to ask, and training the context path changes how it processes the prompt into a format which the rest of the model can understand and respond to.

Hi, could you please explain which is the context path in SD3? Thanks!

**Reply by CodeExplode** [reply]
In SD3 and Flux, there are parts of the model which deal with understanding the prompt (the context path), and parts for understanding the image. I found that in the original SD3, just training the parts which relate to understanding the prompt was enough to teach about any concept. It's just a matter of making sure the model understands what you're asking, and not needing to train how it handles the image segments. It takes a bit of manual coding though.

**Reply by TccccD** [reply]
> In SD3 and Flux, there are parts of the model which deal with understanding the prompt (the context path), and parts for understanding the image. I found that in the original SD3, just training the parts which relate to understanding the prompt was enough to teach about any concept. It's just a matter of making sure the model understands what you're asking, and not needing to train how it handles the image segments. It takes a bit of manual coding though.

This sounds great, could you share the script?

**Reply by CodeExplode** [reply]
There's no script unfortunately, though I've tried it again with SD3M and it seems to be the case as well. Training just the content MLP/Feed Forward parameters seems to train better than full finetuning.
